runner/views/run_api_view.py

```diff
@@ -252,7 +252,8 @@ def post(self, request):
             r.save()
 
             # Copy over the input / output ports from the original Run
-            ports = r.port_set.all()
+            original_run = Run.objects.get(id=original_run_id)
+            ports = original_run.port_set.all()
```

Comment:
good catch -- you could just put `ports = r.port_set.all()` below the loop instead